concent_api/core/views.py

```diff
@@ -329,154 +310,123 @@ def receive_out_of_band(request, _message):
-                    decoded_ack_message.message_task_to_compute,
+                decoded_ack_report_computed_task = load(
+                    serialized_ack_report_computed_task,
                     settings.CONCENT_PRIVATE_KEY,
-                    client_public_key
+                    settings.CONCENT_PUBLIC_KEY,
+                    check_time = False
                 )
             except AttributeError:
                 # TODO: Make error handling more granular when golem-messages adds starts raising more specific exceptions
                 return JsonResponse(
-                    {'error': "Failed to decode TaskToCompute. Message and/or key are malformed or don't match."},
+                    {'error': "Failed to decode ForceReportComputedTask. Message and/or key are malformed or don't match."},
                     status = 400
                 )
+            except InvalidSignature:
+                return JsonResponse({'error': "Wrong message signature"}, status = 400)
 
-            force_report_computed_task = MessageForceReportComputedTask(
-                message_task_to_compute = decoded_ack_message.message_task_to_compute,
-            )
-            message_verdict.message_force_report_computed_task = dump(
-                force_report_computed_task,
-                settings.CONCENT_PRIVATE_KEY,
-                client_public_key
-            )
-            message_verdict.message_ack_report_computed_task = serialized_ack_message
+            force_report_computed_task = ForceReportComputedTask()
+            force_report_computed_task.task_to_compute = decoded_ack_report_computed_task.task_to_compute
+
+            message_verdict.force_report_computed_task = force_report_computed_task
+            message_verdict.ack_report_computed_task   = decoded_ack_report_computed_task
 
             dumped_message_verdict = dump(
                 message_verdict,
                 settings.CONCENT_PRIVATE_KEY,
-                client_public_key
+                requestor_public_key
             )
             store_message(
                 type(message_verdict).__name__,
-                decoded_task_to_compute_message.task_id,
+                decoded_ack_report_computed_task.task_to_compute.compute_task_def['task_id'],
                 dumped_message_verdict
             )
 
-            return message_verdict
+            return dumped_message_verdict
 
-        if last_undelivered_receive_status.type == 'MessageForceReportComputedTask':
-            serialized_force_message = last_undelivered_receive_status.data.tobytes()
+        if last_undelivered_receive_status.type == 'ForceReportComputedTask':
+            serialized_force_report_computed_task = last_undelivered_receive_status.data.tobytes()
             try:
-                decoded_force_message = load(
-                    serialized_force_message,
+                requestor_public_key = get_requestor_public_key(request)
```

Comment:
This line is not necessary. You already have the same line at the beginning of the function so requestor's key is already decoded if it's present.